### ISSUE TITLE
[client] Add fixed linear upscaler

### ISSUE THREAD

**Issue by jfu334** [gnif/LookingGlass#843]
It would be nice to have an option that implements a fixed output scale in the client app. 

Use case: 

- 4k monitor on host
- guest should be run on lower resolution like 1920x1080 for improved performance

Currently it's possible to achieve an exact 2x scale by setting the window size to be twice as large as guest resolution. (Which can be set to any value with tools like CRU.)

This works but is not very comfortable. The dontUpscale option doesn't work, so it's not possible to maximize the guest window. Because the guest resolution is "a bit" smaller than host display size the guest window looks nearly maximized but isn't. 

Not using an exact 2x scale leads to visible distortion in the output.

**Comment by gnif** [comment]
Are you aware that bleeding edge, soon to be B5 has post-processing filters for exactly this use case? We now have both a down-scaler and AMD FSR for upscaling 1080p to 4K

**Comment by jfu334** [comment]
Is there any example/documentation on these? Could find them in the options for the linux client.

For now the easiest solution for me is to set the windows VM to a resolution slightly below 1920x1080 and to do the upscaling in the linux client app so it's readable at 3840x2160.

**Comment by netboy3** [comment]
https://looking-glass.io/docs/bleeding/install/#egl-filters-tab

**Comment by jfu334** [comment]
Thanks for the link. Maybe it was too obvious...

I gave it a try but I can't interact with the EGL filters. 

- I press ScrnLck+O, the Overlay Menu appears. 
- Cursor position based interaction with the menu seems working. Tab headers get highlighted when moving the mouse on them, URLs show a tooltip on hover
- However I can't click any item. Clicks are not recognized, neither left, middle or right. 

There is no related console output.

CORRECTION:

It works after leaving the window (mouse), maximizing the window *and then keeping the mouse in it*. As soon as I leave the window again, the menu stops responding to clicks. 

To the actual cause: The filters look nice and are surely a good thing to have, especially for old games.

However a simple fixed 2x ui scale would be much simpler for me. (Mostly doing UI based workloads)
With that you get the guarantee that there will be no aliasing artifacts.

**Comment by gnif** [comment]
Marking as a feature request for post B5, we are currently feature frozen so it will not be available until then.

As for the mouse issue, please open a new issue and provide exact steps to replicate it, along with what distro you're using, the desktop manager, and versions. Thanks.

**Comment by jfu334** [comment]
Thanks a lot! 

Overall the current unstable version seems to run a lot faster than the stable B4.

**Comment by azampatti** [comment]
Hey there. I'm here testing 5.0.1 and found this about the FSR (which is working FLAWLESSLY for me so finally I can use the VM for 'text reading' since my VM is 1080p and I have a 1440p screen, THANKS!) but I would love to be able to load the preset from the command line.

I've read everything quite a bit and couldn't find a way to do so, did I miss it or currently the only way is to load the saved PRESETS manually each time?

Thanks!
-Aldo

**Comment by gnif** [comment]
@azampatti do not hijack an unrelated issue for your support request. You pollute the thread and make things hard to follow. Go to the LG website and follow the instructions on how to get in touch with the community to ask for help.

**Comment by azampatti** [comment]
> 

Apologies for it, and thanks for your kind comments/advise.

**Comment by gnif** [comment]
This is now available through the option `win:intUpscale` which will limit the client to only up-scaling to integer resolutions (2x, 3x, etc)
